### PR TITLE
Introducing a StorePool class

### PR DESCRIPTION
**Pull request by jadelkhoury**
Introducing a StorePool class

### CODE REVIEW

File: src/store-core/src/main/java/org/eclipse/lyo/store/StorePool.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    private int initialPoolSize;
+    private URI defaultNamedGraphUri;
+    private List<Store> storePool;
+    private List<Store> usedStores;
+    private Object lock = new Object();
+
+    public StorePool (int initialPoolSize, URI defaultNamedGraphUri, URI sparqlQueryEndpoint, URI sparqlUpdateEndpoint, String userName, String password) {
+        this.initialPoolSize = initialPoolSize; 
+        this.defaultNamedGraphUri = defaultNamedGraphUri;
+        this.storePool = new ArrayList<>(this.initialPoolSize);
```
Comment (berezovskyi) [inline src/store-core/src/main/java/org/eclipse/lyo/store/StorePool.java:34]: I would use https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ArrayBlockingQueue.html#ArrayBlockingQueue(int) here.

File: src/store-core/src/main/java/org/eclipse/lyo/store/StorePool.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    public Store getStore() {
+        if (storePool.isEmpty()) {
+            throw new RuntimeException("Maximum pool size reached, no available store connections!");
+        }
+        Store s;
+        synchronized(lock) {
+            if (storePool.isEmpty()) {
+                throw new RuntimeException("Maximum pool size reached, no available store connections!");
+            }
+            s = storePool.remove(storePool.size() - 1);
```
Comment (berezovskyi) [inline src/store-core/src/main/java/org/eclipse/lyo/store/StorePool.java:60]: Lines 57-60 can be replaced with `s=storePool.take()`

File: src/store-core/src/main/java/org/eclipse/lyo/store/StorePool.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import org.eclipse.lyo.store.Store;
+import org.eclipse.lyo.store.StoreFactory;
+
+public class StorePool {
+
+    private int initialPoolSize;
+    private URI defaultNamedGraphUri;
+    private List<Store> storePool;
```
Comment (berezovskyi) [inline src/store-core/src/main/java/org/eclipse/lyo/store/StorePool.java:27]: `BlockingQueue<Store>`

**Comment by jadelkhoury** [comment]
What about this latest patch @berezovskyi ! Class is a lot simpler now.

**Pull request merged**